File: app/dbmodels/AssetPower.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'AssetPower',
    'menu'          => ''
);
$table['fields'] = array(
    'id'               => DBModels::type('increments', 
                                         array('visible' => false)),
    'asset_id'        => DBModels::type('integer', 
                                         array('visible' => false)),
    'power_type_id'    => DBModels::type('integer'),
    'energy_type_id'   => DBModels::type('integer'),
    'is_redondant'     => DBModels::type('boolean'),
);
$table['relationships'] = array(
    'asset'        => array(
        'type'  => 'belongsTo',
        'item'  => 'Asset'),
    'power_type'    => array(
        'type'  => 'belongsTo',
        'item'  => 'PowerType'),
    'energy_type'   => array(
        'type'  => 'belongsTo',
        'item'  => 'EnergyType'),
);

File: fusionglpi/database/asset_power.py
from fusionglpi import db


class AssetPower(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    asset_id = db.Column(db.Integer, db.ForeignKey('asset.id'))
    power_type_id = db.Column(db.Integer, db.ForeignKey('power_type.id'))
    energy_type_id = db.Column(db.Integer, db.ForeignKey('energy_type.id'))
    is_redondant = db.Column(db.Boolean, nullable=False, server_default="0")
